src/chain.cpp

```diff
@@ -35,6 +36,7 @@ CBlockLocator CChain::GetLocator(const CBlockIndex *pindex) const
     std::vector<uint256> vHave;
     vHave.reserve(32);
 
+    READLOCK(cs_chainLock);
     if (!pindex)
```

Comment:
I don't see the use of any WRITELOCK(cs_chainLock) anywhere...if there are readlocks there must also be a few writelocks.

Reply:
There's exactly 1 in SetTip().  I was surprised at this and went over it again myself.  SetTip() is capable of moving the chain forward, rewinding, and hopping to a completely different branch because it inefficiently recreates the entire array every time.